### PR TITLE
feat: filter packages based on glob pattern when filtering by directories

### PR DESCRIPTION
**Pull request by David-Collins**

### CODE REVIEW

**Review by zkochan — CHANGES_REQUESTED** [review]
It is a breaking change. We'll be able to release it only in pnpm v7, if we decide to make this change

File: packages/filter-workspace-packages/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "find-up": "^5.0.0",
     "is-subdir": "^1.1.1",
     "micromatch": "^4.0.2",
+    "multimatch": "^5.0.0",
```
Comment (zkochan) [inline packages/filter-workspace-packages/package.json]: fast-glob should be used, which already a dependency in one of our projects

File: packages/filter-workspace-packages/src/index.ts [hunk 2]
```diff
@@ -55,6 +57,7 @@ export async function filterPackages<T> (
     prefix: string
     workspaceDir: string
     testPattern?: string[]
+    useBetaCli?: boolean
```
Comment (zkochan) [inline packages/filter-workspace-packages/src/index.ts]: I think the option should be more descriptive here. Like `enableDirGlobs`

**Comment by zkochan** [comment]
TBH, I don't understand why this change is needed. If you need to run a command just in one directory, use the `--dir` option.

https://pnpm.io/pnpm-cli#-c-path---dir-path

File: packages/filter-workspace-packages/package.json (hunk 1 above)
Comment (David-Collins) [reply packages/filter-workspace-packages/package.json]: I opted for using micromatch, which is already a dependency in this package and compares the pattern against a given string as opposed to comparing against the directory which I think better fit given that an already built package graph is one of the given arguments in the function where the matching takes place.

**Comment by David-Collins** [comment]
It's not necessarily about running a command in a specific directory, but more about allowing a directory to be a part of a set of filters without it also including any packages that may be subdirectories. 

The best example I can give is that I have a monorepos with an unnamed root package which has scripts I would like to run. I would like to be able to filter a few of the packages within a monorepo as well as some root scripts while not running the scripts of every other package within the monorepo. I would like to avoid duplicate execution of these scripts and some of these root packages may have dependencies on subdirectories so being able to include just the directory as a filter instead of a second pnpm call would work best in this scenario.

**Review by zkochan — APPROVED** [review]

**Pull request merged**